dashboard/src/repositories/clickhouse/weeklyHeatmap.ts

```diff
@@ -0,0 +1,123 @@
+import { clickhouse } from '@/lib/clickhouse';
+import { DateTimeString } from '@/types/dates';
+import { HeatmapMetric, WeeklyHeatmapRow, WeeklyHeatmapRowSchema } from '@/entities/weeklyHeatmap';
+import { BAQuery } from '@/lib/ba-query';
+import { safeSql, SQL } from '@/lib/safe-sql';
+import { QueryFilter } from '@/entities/filter';
+
+function getBaseAggregation(metric: HeatmapMetric) {
+  switch (metric) {
+    case 'pageviews':
+      return safeSql`count()`;
+    case 'unique_visitors':
+      return safeSql`uniq(visitor_id)`;
+    case 'sessions':
+      return safeSql`uniq(session_id)`;
+    case 'bounce_rate':
+      return null as unknown as ReturnType<typeof safeSql>;
+    case 'pages_per_session':
+      return null as unknown as ReturnType<typeof safeSql>;
+    case 'session_duration':
+      return null as unknown as ReturnType<typeof safeSql>;
+  }
+}
+
+export async function getWeeklyHeatmap(
+  siteId: string,
+  startDate: DateTimeString,
+  endDate: DateTimeString,
+  metric: HeatmapMetric,
+  queryFilters: QueryFilter[],
+): Promise<WeeklyHeatmapRow[]> {
+  const filters = BAQuery.getFilterQuery(queryFilters);
+
+  if (metric === 'bounce_rate' || metric === 'pages_per_session' || metric === 'session_duration') {
+    // Build from session_data per hour bucket then aggregate by weekday/hour across the range
+    const query = safeSql`
+      WITH session_data AS (
+        SELECT
+          session_id,
+          toDayOfWeek(timestamp) as weekday,
+          toHour(timestamp) as hour,
+          count() as page_count,
+          if(count() > 1, dateDiff('second', min(timestamp), max(timestamp)), 0) as duration_seconds
+        FROM analytics.events
+        WHERE site_id = {site_id:String}
+          AND timestamp BETWEEN {start:DateTime} AND {end:DateTime}
+          AND ${SQL.AND(filters)}
```

Comment:
This should probably not look at cwv, custom events, etc. and only do on page_views